### ISSUE TITLE
sql: column does not exists error for statement with alias inside UDF

### ISSUE THREAD

**Issue by mgartner** [cockroachdb/cockroach#104927]
Consider the table:

```sql
CREATE TABLE t (
  i INT,
  s STRING
);

INSERT INTO t VALUES (1, 'foo');
```

Running this statement:

```sql
SELECT json_agg(r) FROM (
  SELECT i, s
  FROM t
) AS r;
```

Produces:

```
         json_agg
--------------------------
  [{"i": 1, "s": "foo"}]
(1 row)
```

Running the same statement inside a UDF:

```sql
CREATE FUNCTION f() RETURNS TEXT LANGUAGE SQL AS $$
  SELECT json_agg(r) FROM (
    SELECT i, s
    FROM t
  ) AS r
$$;

SELECT f();
```

Produces an error:

```
ERROR: column "r" does not exist
SQLSTATE: 42703
```

Jira issue: CRDB-28779

**Comment by mikeCRL** [comment]
@mgartner Since this was raised and resolved across June and July, am I correct to assume this should not be surfaced in v23.2 docs? I.e. it is no longer a Known Limitation as of v23.2 GA?

**Comment by mgartner** [comment]
@mikeCRL Correct, this is fixed in v23.2.
